analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,924 @@
+
+.. data:: scope
+
+   **Required** Scope of the variable, "visit" or "page"
+
+
+.. warning::
+
+    Custom Variable with scope set on "visit" will be saved for visit, you don't need to save it every page.
+
+.. warning::
+
+    Index is separate for each variable scope.
+
+Example of usage::
+
+    _paq.push(["setCustomVariable", 1, "AspectRatio", "16:9", "visit"]);
+
+Removing Custom Variable
+````````````````````````
+To remove custom variable you can use ``deleteCustomVariable`` function::
+
+    _paq.push(["deleteCustomVariable", index, scope]);
+
+.. data:: index
+
+    **Required** Number from 1 to 5 where variable is stored
+
+.. data:: scope
+
+    **Required** Scope of the variable, "visit" or "page"
+
+Example of usage::
+
+    _paq.push(["deleteCustomVariable", 1, "visit"]);
+
+Accessing Custom Variable
+`````````````````````````
+.. todo::
+It would be nice to have some examples of returned data.
+
+You can access custom variables by providing function that will use ``getCustomVariable`` function::
+
+    _paq.push([ function() {
+        var customVariable = this.getCustomVariable(index, scope );
+    }]);
+
+.. function:: getCustomVariable(index, scope)
+
+    :param number index: **Required** Number from 1 to 5 where variable is stored
+
+    :param string scope: **Required** Scope of the variable, "visit" or "page"
+
+Example of usage::
+
+    _paq.push([ function() {
+        var customVariable = this.getCustomVariable(1, "visit" );
+        console.log(customVariable);
+    }]);
+
+Custom Dimensions
+^^^^^^^^^^^^^^^^^
+
+Tracking Custom Dimension
+`````````````````````````
+If you want to set custom dimension to use it in tracking functions use ``setCustomDimension`` function::
+
+    _paq.push(["setCustomDimension", customDimensionID, customDimensionValue]);
+
+.. data:: customDimensionID
+
+    **Required** Id of dimension
+
+.. data:: customDimensionValue
+
+    **Required** Value of Custom Dimension
+
+.. warning::
+
+    When you set Custom Dimension that value will be used in all tracking requests within page load.
+
+Example of usage::
+
+    _paq.push(["setCustomDimension", 3, "loginStatus"]);
+
+
+Retrieving Custom Dimension
+```````````````````````````
+.. todo::
+It would be nice to have some examples of returned data.
+
+You can access custom dimension by providing function that will use ``getCustomDimension`` function::
+
+    _paq.push([ function() {
+        var customDimension = this.getCustomDimension(index);
+    }]);
+
+.. function:: getCustomDimension(index, scope)
+
+    :param number index: **Required** Index of custom dimension
+
+Example of usage::
+
+    _paq.push([ function() {
+        var customDimension = this.getCustomDimension(1);
+        console.log(customDimension);
+    }]);
+
+Content Tracking
+^^^^^^^^^^^^^^^^
+
+Content Tracking is an useful feature when you want to measure performance of your website or app. For example you can check
```

Comment:
Typo: "an useful" => "a useful" I think, but I would skip such adjectives and just explain what it does:

Content Tracking tracks how many times specific elements were rendered/visible. It can be used to measure if ad placement was visible or if user have seen end of article.

You mentioned there also file download. I thought that isn't part of this module, but if it is - then such example is also OK.